operators/src/source/gdal_source.rs

```diff
@@ -372,6 +374,155 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+        sorted_time_steps.sort_by(|a, b| a.end().partial_cmp(&b.start()).unwrap());
+
+        GdalMetadataFixedTimes {
+            time_steps: sorted_time_steps,
+            params,
+            result_descriptor,
+            time_placeholders,
+        }
+    }
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for GdalMetadataFixedTimes
+{
+    async fn loading_info(&self, query: RasterQueryRectangle) -> Result<GdalLoadingInfo> {
+        let minimum_step = TimeStep {
+            granularity: TimeGranularity::Millis,
+            step: 1,
+        };
+
+        let sorted_time_steps = if self
+            .time_steps
+            .windows(2)
+            .all(|w| w[0].end() <= w[1].start())
+        {
+            self.time_steps.clone()
+        } else {
+            let mut sorted_time_steps = self.time_steps.clone();
+            sorted_time_steps.sort_by(|a, b| a.end().partial_cmp(&b.start()).unwrap());
+            sorted_time_steps
+        };
+
+        let mut time_steps_no_gaps = sorted_time_steps.clone();
+
+        for (index, time) in sorted_time_steps.iter().enumerate() {
+            if index < sorted_time_steps.len() - 1 {
+                if time.is_instant() {
+                    time_steps_no_gaps.push(TimeInterval::new(
+                        (time.end() + minimum_step)?,
+                        sorted_time_steps[index + 1].start(),
+                    )?);
+                } else {
+                    time_steps_no_gaps.push(TimeInterval::new(
+                        time.end(),
+                        sorted_time_steps[index + 1].start(),
+                    )?);
+                }
+            }
+        }
+
+        time_steps_no_gaps.sort_by(|a, b| a.start().partial_cmp(&b.start()).unwrap());
+
+        let valid_duration = TimeInterval::new(
+            time_steps_no_gaps[0].start(),
+            time_steps_no_gaps[time_steps_no_gaps.len() - 1].start(),
+        )?;
+
+        let parts = match valid_duration.intersect(&query.time_interval) {
+            Some(time_interval) => {
+                let mut loading_info_parts: Vec<GdalLoadingInfoPart> = Vec::new();
+
+                if query.time_interval.start() < time_interval.start() {
+                    let gap =
+                        TimeInterval::new(query.time_interval.start(), time_interval.start())?;
+
+                    loading_info_parts.push(
+                        self.params
+                            .replace_time_placeholders(&self.time_placeholders, gap)
+                            .map(|loading_info_part_params| GdalLoadingInfoPart {
+                                time: gap,
+                                params: loading_info_part_params,
+                            })?,
+                    );
+                }
+                for time_position in &time_steps_no_gaps {
+                    if let Some(step_interval) = time_interval.intersect(time_position) {
```

Comment:
do you really have to check for intersction every time?

Reply:
This is to create the intersect `TimeInterval`, which is then compared with the original `time_steps` to check if it should create valid or no-data tile.